### DISCUSSION TITLE
Um ... MetricPoint Um.. wtf?

### DISCUSSION THREAD

**Discussion by eaglestorm** [open-telemetry/opentelemetry-dotnet#5915] [Q&A]
Created a customer exporter to write to a CSV file and happened to look at the source code for the MetricPoint class.

ThrowNotSupportedMetricTypeException is used to throw a NotSupportedException

Um using an exception like this is an indication of badly designed code, hopefully this is already know and its going to be fixed in a future updates.  Its also bad for performance if these exceptions get triggered at runtime.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MetricPoint/MetricPoint.cs#L1096 

I considered posting to The daily WTF as yes its that bad so hopefully this is planed to be fixed.

**Comment by cijothomas** [comment]
Not clear what is the actual ask/issue.
